src/components/LabelPicker/LabelPicker.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+
+export type Label = {
+    /** the display name to be shown to the user */
+    name: string;
+    /** state of the label: selected or not */
+    isSelected: boolean;
+    /** the number of occurrences will be shown next to the name */
+    count?: number;
+};
+
+export interface Props<B = HTMLButtonElement> {
+    /** a list of available labels and their attributes */
+    labels: Label[];
+    /** a button like element that supports onClick handler to be used as the trigger */
+    children: React.ReactElement<React.ButtonHTMLAttributes<B>>;
+    /** callback to be called when the state of a checkbox changes */
+    onChange: (Label) => void;
+    /** callback to add new label */
+    onAdd: (name: string) => void;
+    /** callback when clicking the Done button */
+    onDone?: () => void;
+    /** callback on outer click */
+    onCancel?: () => void;
```

Comment:
What do you think about renaming prop `onCancel` -> `onClose`?

Reply:
Not sure. clicking on `Done` should close the dialog too. We can call it `onOuterClick` maybe

Reply:
after f2f discussion decided to keep only 1 callback `onClose` that will be called regardless where the user clicked to close the component (trigger button, done button or outer click)